File: 0264/264.py
```
Write a program to find the n-th ugly number.

Ugly numbers are positive numbers whose prime factors only include 2, 3, 5. 

Example:

Input: n = 10
Output: 12
Explanation: 1, 2, 3, 4, 5, 6, 8, 9, 10, 12 is the sequence of the first 10 ugly numbers.
```
class Solution:
    def nthUglyNumber(self, n: int) -> int:
        ugly =[1]
        p2,p3,p5=0,0,0
        for i in range(n):
            ug2=ugly[p2]*2
            ug3=ugly[p3]*3
            ug5=ugly[p5]*5
            ug = min(ug2,min(ug3,ug5))
            ugly.append(ug)
            if ug2==ug:
                p2+=1
            if ug3 == ug:
                p3+=1
            if ug5==ug:
                p5+=1
        return ugly[n-1]
                

File: 0264/264.java
/*
Write a program to find the n-th ugly number.

Ugly numbers are positive numbers whose prime factors only include 2, 3, 5. 

Example:

Input: n = 10
Output: 12
Explanation: 1, 2, 3, 4, 5, 6, 8, 9, 10, 12 is the sequence of the first 10 ugly numbers.
Note:  

1 is typically treated as an ugly number.
n does not exceed 1690.
*/
class Solution {
    public int nthUglyNumber(int n) {
        int[] help = new int[n];
        int two = 0;
        int three = 0;
        int five = 0;
        help[0]=1;
        int min_val = 0;
        for(int i=1;i<n;i++){
            min_val = Math.min(Math.min(help[two]*2,help[three]*3),help[five]*5);
            if(help[two]*2==min_val){
                two++;
            }
            if(help[three]*3==min_val){
                three++;
            }
            if(help[five]*5==min_val){
                five++;
            }
            help[i]=min_val;
        }
        return help[n-1];
    }
}